src/java.base/windows/classes/sun/nio/fs/WindowsFileStore.java

```diff
@@ -44,6 +52,9 @@
     private final int volType;
     private final String displayName;   // returned by toString
 
+    private boolean hasHashCode = false; // as hashCode can be any int
```

Comment:
This is buggy. It would be simpler to drop hasHashCode and just race when hash is 0.